### DISCUSSION TITLE
bug:Can't parse go code correctly

### DISCUSSION THREAD

**Discussion by jingyuexing** [vuejs/vitepress#2853] [Q&A]
### Describe the bug

```
[plugin:vite:vue] Error parsing JavaScript expression: Unexpected token, expected "," (1:12)
F:/Code/vuepress-demo/src/module/i18n.md:129:14
128|  		this.Ctx.SetCookie("lang", curLang.Lang, 1<<31-1, "/")
129|  	}
130|  	restLangs \:= make([]*langType, 0, len(langTypes)-1)
   |             ^
131|  	for _, v := range langTypes {
132|  		if lang != v.Lang {
```

### Reproduction

```md

```go
// setLangVer sets site language version.
func (this *baseRouter) setLangVer() bool {
	isNeedRedir := false
	hasCookie := false

	// 1. Check URL arguments.
	lang := this.Input().Get("lang")

	// 2. Get language information from cookies.
	if len(lang) == 0 {
		lang = this.Ctx.GetCookie("lang")
		hasCookie = true
	} else {
		isNeedRedir = true
	}

	// Check again in case someone modify by purpose.
	if !i18n.IsExist(lang) {
		lang = ""
		isNeedRedir = false
		hasCookie = false
	}

	// 3. Get language information from 'Accept-Language'.
	if len(lang) == 0 {
		al := this.Ctx.Request.Header.Get("Accept-Language")
		if len(al) > 4 {
			al = al[:5] // Only compare first 5 letters.
			if i18n.IsExist(al) {
				lang = al
			}
		}
	}

	// 4. Default language is English.
	if len(lang) == 0 {
		lang = "en-US"
		isNeedRedir = false
	}

	curLang := langType{
		Lang: lang,
	}

	// Save language information in cookies.
	if !hasCookie {
		this.Ctx.SetCookie("lang", curLang.Lang, 1<<31-1, "/")
	}
	restLangs := make([]*langType, 0, len(langTypes)-1)
	for _, v := range langTypes {
		if lang != v.Lang {
			restLangs = append(restLangs, v)
		} else {
			curLang.Name = v.Name
		}
	}

	// Set language properties.
	this.Lang = lang
	this.Data["Lang"] = curLang.Lang
	this.Data["CurLang"] = curLang.Name
	this.Data["RestLangs"] = restLangs

	return isNeedRedir
}
```

### Expected behavior

Should be correctly parsing code blocks, not parsing templates

### System Info

```
"vitepress": "1.0.0-alpha.76"
```

### Validations

- [X] Check if you're on the [latest VitePress version](https://github.com/vuejs/vitepress/releases/latest).
- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Did you miss ` ``` ` after that code block? It appears to be working fine for me: https://stackblitz.com/edit/vite-urnofi?file=docs/index.md

**Comment by jingyuexing** [comment]
> 



> Did you miss `` ``` `` after that code block? It appears to be working fine for me: https://stackblitz.com/edit/vite-urnofi?file=docs/index.md

nop,i'm make sure

**Comment by jingyuexing** [comment]
> Did you miss `` ``` `` after that code block? It appears to be working fine for me: https://stackblitz.com/edit/vite-urnofi?file=docs/index.md

![image](https://github.com/vuejs/vitepress/assets/19589872/0aeb2f5f-e9a0-4706-984f-271c6e5cfd35)

**Comment by jingyuexing** [comment]
> Did you miss `` ``` `` after that code block? It appears to be working fine for me: https://stackblitz.com/edit/vite-urnofi?file=docs/index.md

Now wrongly changed location

**Comment by jingyuexing** [comment]
> Did you miss `` ``` `` after that code block? It appears to be working fine for me: https://stackblitz.com/edit/vite-urnofi?file=docs/index.md

I posted the full text of the document, and it has begun to report errors

**Comment by jingyuexing** [comment]
fantasy！

**Comment by jingyuexing** [comment]
https://stackblitz.com/edit/vite-3etpsv?file=docs%2Findex.md

**Comment by zonemeen** [comment]
<img width="281" alt="image" src="https://github.com/vuejs/vitepress/assets/44596995/1bb33698-826d-4c2e-82a1-fbc3c0ea0280">

The error was caused by this line. What does it do?

**Comment by jingyuexing** [comment]
> <img alt="image" width="281" src="https://user-images.githubusercontent.com/44596995/239685462-1bb33698-826d-4c2e-82a1-fbc3c0ea0280.png">
> The error was caused by this line. What does it do?

template,is the template syntax of the backend framework

**Comment by zonemeen** [comment]
Does it render anything, or just display characters?

**Comment by jingyuexing** [comment]
> Does it render anything, or just display characters?

this is beego syntax，not vue

**Comment by zonemeen** [comment]
Could you provide a full reproduction?

**Comment by jingyuexing** [comment]
> Could you provide a full reproduction?

[beego](https://github.com/beego/beedoc/tree/master/zh-CN)
